larky/src/main/java/com/verygood/security/larky/modules/VaultModule.java

```diff
@@ -0,0 +1,120 @@
+package com.verygood.security.larky.modules;
+
+import com.google.common.collect.Iterators;
+import com.verygood.security.larky.modules.vgs.vault.DefaultVault;
+import com.verygood.security.larky.modules.vgs.vault.spi.LarkyVault;
+import com.verygood.security.larky.modules.vgs.vault.NoopVault;
+import net.starlark.java.annot.Param;
+import net.starlark.java.annot.ParamType;
+import net.starlark.java.annot.StarlarkBuiltin;
+import net.starlark.java.annot.StarlarkMethod;
+import net.starlark.java.eval.EvalException;
+import net.starlark.java.eval.NoneType;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.ServiceLoader;
+
+
+@StarlarkBuiltin(
+        name = "vault",
+        category = "BUILTIN",
+        doc = "Overridable Vault API in Larky")
+public class VaultModule implements LarkyVault {
+
+    public static final VaultModule INSTANCE = new VaultModule();
+    public static final String PROPERTY_NAME = "larky.modules.vgs.vault.enableInMemoryVault";
+
+    private LarkyVault vault;
+
+    public VaultModule() {
+
+        ServiceLoader<LarkyVault> loader = ServiceLoader.load(LarkyVault.class);
+        Iterator<LarkyVault> vaultIterator = loader.iterator();
```

Comment:
as far as i can see, we do not anticipate huge amount of services, so it is safe to convert this iterator to list. (+ it will be garbage collected after class initialization).
Please, consider doing so, it will result in lessen code cognitive load as 
line 37: `if (!vaultIterator.hasNext()) {` will transform in `vaultServices.isEmpty()`
line 41: `if (vaultIterator.hasNext())` -> `if (vaultServices.size() != 1)`
lines 43-44: `"VaultModule expecting only 1 vault provider of type LarkyVault, found " + (1 + Iterators.size(vaultIterator))` -> `"VaultModule expecting only 1 vault provider of type LarkyVault, found " + vaultServices.size()`